06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * A class to create and search for indexes
+ * @class InvertedIndex
+ */
+class InvertedIndex {
+
+  /**
+   * Creates an instance of InvertedIndex.
+   * @memberOf InvertedIndex
+   */
+  constructor() {
+    this.index = {};
+    this.documentWholeText = {};
+    this.documentWholeTitle = {};
+    this.documentRange = {};
+    this.all = false;
+    this.get = false;
+    this.search = false;
+  }
+
+  /**
+   * To validate the content of the uploaded JSON files
+   * @static
+   * @param {Object} jsonFile - Parsed JSON file
+   * @returns {Boolean} True or False
+   * @memberOf InvertedIndex
+   */
+  static validateContent(jsonFile) {
+    let isValid = true;
+    jsonFile.forEach((book) => {
+      if (typeof book.title === 'undefined'
+      || typeof book.text === 'undefined') {
+        isValid = false;
+      }
+    });
+    return isValid; // use flag instead
+  }
+
+  /**
+   * Indexes JSON file
+   * @param {Object} fileContent - JSON file uploaded into app to be indexed
+   * @param {String} fileName - Name of JSON file
+   * @returns {Object} - Words as keys and a value of a array with
+   * their repective books
+   * @memberOf InvertedIndex
+   */
+  createIndex(fileContent, fileName) {
+    const titleObj = {};
+    const textObj = {};
+    let count = 1;
+    fileContent.forEach((book) => {
+      titleObj[count] = book.title;
+      textObj[count] = book.text;
+      count += 1;
+    });
+    Object.keys(textObj).forEach((words) => {
+      textObj[words] = textObj[words].replace(/'\w+\s/g, ' ')
+      .replace(/[.,/#!+$%^&@*?;:'{}=\-_`~()]/g, '').trim().toLowerCase()
+      .split(' ');
+    });
+    this.documentWholeText[fileName] = this.populateIndex(textObj);
+    this.documentWholeTitle[fileName] = titleObj;
+
+    return this.documentWholeText[fileName];
+  }
```

Comment:
Something seems a little different here. Is your a createIndex() doing a lot more than just creating the Index?